crm_stage_type/models/crm_lead.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import SUPERUSER_ID, api, fields, models
+
+
+class Lead(models.Model):
+    _inherit = "crm.lead"
+
+    stage_id = fields.Many2one(
+        domain="[('team_id', 'in', [team_id, False]), "
+        "('lead_type', 'in', [type, 'both'])]"
+    )
+
+    @api.model
+    def _read_group_stage_ids(self, stages, domain, order):
+        ctx_type = self.env.context.get("default_type")
+        stages = super(Lead, self)._read_group_stage_ids(stages, domain, order)
+        search_domain = [("id", "in", stages.ids)]
+        if ctx_type:
+            search_domain += [("lead_type", "in", [ctx_type, "both"])]
+        stage_ids = stages._search(
+            search_domain, order=order, access_rights_uid=SUPERUSER_ID
+        )
+        return stages.browse(stage_ids)
+
+    def _stage_find(self, team_id=False, domain=None, order="sequence"):
+        # check whether we should try to add a condition on type
+        if domain and not any(
```

Comment:
hhhmmm.... @ValentinVinagre what do you think? might a bug be introduced here as @skeller1 is suggesting? Thx!

Reply:
crm.lead model is calling _stage_find() without a domain
[see here in 13.0](https://github.com/odoo/odoo/blob/2100f07cf2eea3c813ea1ee702d25b0beb3a1e8e/addons/crm/models/crm_lead.py#L879)
[see here in 12.0 ](https://github.com/odoo/odoo/blob/198ab837e466f3155ea1c3b35811b981b1c1e537/addons/crm/models/crm_lead.py#L777)

The current crm_stage_type 12.0 implementation should raise an error if we pass no domain:
`Traceback (most recent call last):
TypeError: 'NoneType' object is not iterable
`
Fixed it here defining domain as an empty list (iteratable) if it is None

Reply:
Hi @skeller1 you add `domain = domain or []`?
This fixes the problem, I think it is a correct solution.
Another solution could be:
 `[term for term in (domain or []) if len(term) == 3 if term[0] == "lead_type"]` 
but since that variable can be used later I think the applied solution is correct.
👍